docs/development/component-checklist.md

```diff
@@ -0,0 +1,146 @@
+
+   You should use the [secrets manager](secrets_management.md) for the management of any kind of credentials.
+   This makes sure that credentials rotation works out-of-the-box without you requiring to think about it.
+   Generally, do not use client certificates (see [security section](#security)).
+
+6. **Consider hibernation when calculating replica count** ([example](https://github.com/gardener/gardener/blob/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/pkg/operation/botanist/kubescheduler.go#L36))
+
+   Shoot clusters can be [hibernated](../usage/shoot_hibernate.md) meaning that all control plane components in the shoot namespace in the seed cluster are scaled down to zero and all worker nodes are terminated.
+   If your component runs in the seed cluster then you have to consider this case and provide the proper replica count.
+   There is a utility function available (see example).
+
+7. **Ensure task dependencies are as precise as possible in shoot flows** ([example 1](https://github.com/gardener/gardener/blob/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go#L508-L512), [example 2](https://github.com/gardener/gardener/blob/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/pkg/gardenlet/controller/shoot/shoot/reconciler_delete.go#L368-L372))
+
+   Only define the minimum of needed dependency tasks in the [shoot reconciliation/deletion flows](https://github.com/gardener/gardener/tree/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/pkg/gardenlet/controller/shoot/shoot). 
+
+## Security
+
+1. **Use a [dedicated `ServiceAccount`](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) and disable auto-mount** ([example](https://github.com/gardener/gardener/blob/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/pkg/operation/botanist/component/metricsserver/metrics_server.go#L145-L151))
+
+   Components which need to talk to the API server of their runtime cluster must always use a dedicated `ServiceAccount` (do not use `default`) which `automountServiceAccountToken` set to `false`.
+   This makes `gardener-resource-manager`'s [TokenInvalidator](https://github.com/gardener/gardener/blob/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/docs/concepts/resource-manager.md#tokeninvalidator) invalidating the static token secret and its [`ProjectedTokenMount` webhook](https://github.com/gardener/gardener/blob/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/docs/concepts/resource-manager.md#auto-mounting-projected-serviceaccount-tokens) injecting a projected token automatically.
+
+2. **Use shoot access tokens instead of a client certificates** ([example](https://github.com/gardener/gardener/blob/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/pkg/operation/botanist/component/kubescheduler/kube_scheduler.go#L227-L229))
```

Comment:
Nit: when reading "shoot access token" devs might think, they can only use it for shoot clusters. But this will also be possible for components running in the garden runtime cluster talking to the virtual garden cluster.

Reply:
Well, for now, this is the status quo. I would rather keep it for now to not add information here which has not become reality yet. We can adapt this document when this feature (GRM for virtual garden) is supported by `gardener-operator`.

Reply:
Agreed